### DISCUSSION TITLE
Usage of slots

### DISCUSSION THREAD

**Discussion by pdanpdan** [kennyfrc/cami.js#5] [General]
Hello. Is there a way to use slots/html content provided inside the component?

**Comment by kennyfrc** [comment]
Hi @pdanpdan! We cannot use slots as Cami inherits from the Light DOM and does not use the Shadow DOM / Shadow Root.

I found a few issues with the shadow DOM:
- The style isolation confuses some folks, and most people are more comfortable with just working with a global CSS stylesheet
- The shadow DOM introduces FOUC (flash of unstyled content). So if you have static HTML around a component that uses the shadow DOM, there will be some content shifts that you have [to hack around](https://stackoverflow.com/questions/62683430/how-to-stop-fouc-from-happening-with-native-web-components).
- Shadow DOM is additional overhead for the browser, so it loads a bit slower than the Light DOM

For users who want to isolate styles but want to use web components (with no shadow DOM), you can add isolated styles with [goober](https://github.com/cristianbote/goober), which generates class names and allows you to insert it into the html tagged template.

If slots are important for you, [Lit](https://lit.dev/) might be a better choice.

**Comment by pdanpdan** [comment]
In fact I was thinking more of creating shells around content. That would render the content of the element wrapped in some other html
